pkg/seven/state/builder.go

```diff
@@ -84,161 +86,198 @@ func paddingUpstream(upstream *v1.Upstream, currentUpstream *v1.Upstream) {
 // NewRouteWorkers make routeWrokers group by service per CRD
 // 1.make routes group by (1_2_3) it may be a map like map[1_2_3][]Route;
 // 2.route is listenning Event from the ready of 1_2_3;
-func NewRouteWorkers(routes []*v1.Route) RouteWorkerGroup {
+func NewRouteWorkers(ctx context.Context, routes []*v1.Route, wg *sync.WaitGroup, errorChan chan CRDStatus) RouteWorkerGroup {
 	rwg := make(RouteWorkerGroup)
 	for _, r := range routes {
-		quit := make(chan Quit)
-		rw := &routeWorker{Route: r, Quit: quit}
+		rw := &routeWorker{Route: r, Ctx: ctx, Wg: wg, ErrorChan: errorChan}
 		rw.start()
 		rwg.Add(*r.ServiceName, rw)
 	}
 	return rwg
 }
 
 // 3.route get the Event and trigger a padding for object,then diff,sync;
-func (r *routeWorker) trigger(event Event) error {
-	defer close(r.Quit)
+func (r *routeWorker) trigger(event Event) {
+	var errNotify error
+	defer func(err error) {
+		if err != nil {
+			r.ErrorChan <- CRDStatus{Id: "", Status: "failure", Err: err}
+		}
+		r.Wg.Done()
+	}(errNotify)
```

Comment:
Here you should use `errNotify` as a Closure variable rather than function parameter, since the parameters deduction of deferred function happen when the defer statement runs (runtime.deferproc), not the function runs(`runtime.deferreturn`)

Reply:
Done